medical-system/src/components/shared/Table.tsx

```diff
@@ -127,13 +89,13 @@ export function Table() {
                 ))}
                 <td className="border border-slate-200 p-2 flex justify-center items-center gap-4">
                   <button
-                    onClick={() => handleDelete(row.original.id)}
+                    onClick={() => onDelete(row.original.id)}
```

Comment:
Tutaj możesz spróbować zastosować domknięcie zamiast inline funkcje, to dobry trening.

Tutaj robisz:
`onClick={handleDelete(row.original.id)}`

A poza JSX robisz `const handleDelete = (id: string) => () => { ciało funkcji }`

Dzieki temu, zamiast robić funkcje anonimową () => w JSX, tworzysz funkcje poza. Jej wywołanie zwraca funkcje ktora sie wklei pod onClickiem i bedzie sobie działać :P funkcja zwroci przygotowaną funkcję do uruchamiania kliknieciem

Reply:
Zwróć uwagę że tam są dwa znaki => co oznacza ze definiujesz funkcje ktora zwroci funkcje. Ta zagniezdzona funkcja zapamieta sobie ID i bedzie na tej podstawie robic rzeczy potrzebne do delete